# Are colder countries more prosperous?  
  
Colder countries generally appear more prosperous. Lets examine that with some data.  
  
Data Sources: https://pkgstore.datahub.io/core/country-codes/country-codes_zip/data/f9df99397e4c97cf45a7f174da14c073/country-codes_zip.zip  
              https://databank.worldbank.org/data/download/catalog/cckp_historical_data_0.xls  
              https://www.kaggle.com/rohan0301/unsupervised-learning-on-country-data  
                
The data sets were combined on country codes and names.  
    
Please use **Show/Hide** Code button below to see the code              

In [85]:
from IPython.display import HTML

HTML('''<script>
code_show=true; 
function code_toggle() {
 if (code_show){
 $('div.input').hide();
 } else {
 $('div.input').show();
 }
 code_show = !code_show
} 
$( document ).ready(code_toggle);
</script>
<form action="javascript:code_toggle()"><input type="submit" value="Show/Hide Code"></form>''')

In [57]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
%matplotlib inline

In [58]:
import chart_studio.plotly as py
from plotly import __version__
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot
import cufflinks as cf
init_notebook_mode(connected=True)
cf.go_offline()

In [59]:
data = pd.read_csv('countrydata.csv')

In [61]:
del data['Unnamed: 0']

In [62]:
data

country  child_mort  exports  health  imports  income  inflation  \
0    Afghanistan        90.2     10.0    7.58     44.9    1610       9.44   
1        Albania        16.6     28.0    6.55     48.6    9930       4.49   
2        Algeria        27.3     38.4    4.17     31.4   12900      16.10   
3         Angola       119.0     62.3    2.85     42.9    5900      22.40   
4      Argentina        14.5     18.9    8.10     16.0   18700      20.90   
..           ...         ...      ...     ...      ...     ...        ...   
129      Uruguay        10.6     26.3    8.35     25.4   17100       4.91   
130   Uzbekistan        36.3     31.7    5.81     28.5    4240      16.50   
131      Vanuatu        29.2     46.6    5.25     52.7    2950       2.62   
132        Yemen        56.3     30.0    5.18     34.4    4480      23.60   
133       Zambia        83.1     37.0    5.89     30.9    3280      14.00   

     life_expec  total_fer   gdpp   temp code  
0          56.2       5.82    553  12.92  AFG  
1          76.3       1.65   4090  11.27  ALB  
2          76.5       2.89   4460  22.58  DZA  
3          60.1       6.16   3530  21.51  AGO  
4          75.8       2.37  10300  14.22  ARG  
..          ...        ...    ...    ...  ...  
129        76.4       2.08  11900  17.57  URY  
130        68.8       2.34   1380  12.52  UZB  
131        63.0       3.50   2970  23.84  VUT  
132        67.5       4.67   1310  23.16  YEM  
133        52.0       5.40   1460  21.58  ZMB  

[134 rows x 12 columns]

## GDP per capita vs Tempareature (Interactive plot)

In [79]:

fig = px.scatter(data, x="temp", y="gdpp", hover_data=['country'], width=1000, height=600)

fig.show()

There seems to be some truth to the observation that colder countries tend to be more prosperies. The warmer countries that have relatively high gdpp are mostly oil rich Gulf countries and Australia

## Fertility rate vs tempareature with color and size of marker representing GDP/capita

In [84]:
px.scatter(data, x="temp", y="total_fer", hover_data=['country'], size='gdpp',color='gdpp', width=1000, height=600)

We see a cluster of prosperous countries with fertility rate around 2.1 which is the replacement fertility rate. These countries also have high gdpp and are also quite cold. 